### PR TITLE
Add serdes bbe

### PR DESCRIPTION
**Pull request by MohamedSabthar**
Related issue: [#3075](https://github.com/ballerina-platform/ballerina-standard-library/issues/3075)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/index.json [hunk 1]
```diff
@@ -2532,6 +2532,21 @@
...
+    {
+        "title": "SerDes",
+        "column": 3,
+        "category": "Working over the network",
+        "samples": [
+            {
+                "name": "Serialization and Deserialization",
+                "url": "serdes-simple",
+                "verifyBuild": true,
+                "verifyOutput": false,
```
Comment (niveathika) [inline examples/index.json:2544]: Why is output not verified?

Comment (MohamedSabthar) [reply examples/index.json:2544]: Need to verify, will change the flag

Comment (MohamedSabthar) [reply examples/index.json]: Updated.

File: examples/index.json [hunk 2]
```diff
@@ -2532,6 +2532,21 @@
...
+        "title": "SerDes",
+        "column": 3,
+        "category": "Working over the network",
+        "samples": [
+            {
+                "name": "Serialization and Deserialization",
+                "url": "serdes-simple",
+                "verifyBuild": true,
+                "verifyOutput": true,
+                "disablePlayground": true,
```
Comment (ThisaruGuruge) [inline examples/index.json:2545]: I think we can enable this.

File: examples/serdes-simple/serdes_simple.metatags [hunk 3]
```diff
@@ -0,0 +1,2 @@
+description: This Ballerina sample demonstrates how to perform serialization and deserialization using the serdes module.
+keywords: ballerina, ballerina by example, bbe, serialization, deserialization, serdes
```
Comment (ThisaruGuruge) [inline examples/serdes-simple/serdes_simple.metatags:2]: This maybe more specific for the BBE

```suggestion
keywords: ballerina, ballerina by example, proto3, serialization, deserialization, serdes
```

File: examples/index.json (hunk 2 above)
Comment (MohamedSabthar) [reply examples/index.json:2545]: "disablePlayground": false ?

File: examples/serdes-simple/serdes_simple.metatags (hunk 3 above)
Comment (MohamedSabthar) [reply examples/serdes-simple/serdes_simple.metatags]: Suggestion commited.

File: examples/index.json (hunk 2 above)
Comment (MohamedSabthar) [reply examples/index.json]: updated.

**Review by ThisaruGuruge — APPROVED** [review]

File: examples/serdes-simple/serdes_simple.bal [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import ballerina/io;
+import ballerina/serdes;
+
+// Define a type which is a subtype of anydata.
```
Comment (praneesha) [inline examples/serdes-simple/serdes_simple.bal:4]:
```suggestion
// Define a type, which is a subtype of anydata.
```

File: examples/serdes-simple/serdes_simple.bal [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+type Student record {
+    int id;
+    string name;
+    decimal fees;
+};
+
+public function main() returns error? {
+
+    // Assign the value to the variable
```
Comment (praneesha) [inline examples/serdes-simple/serdes_simple.bal:13]:
```suggestion
    // Assign the value to the variable.
```

File: examples/serdes-simple/serdes_simple.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Serialization and Deserialization
+
+The serdes module helps to serialize and deserialize subtypes Ballerina `anydata`.
```
Comment (praneesha) [inline examples/serdes-simple/serdes_simple.md:3]:
```suggestion
The `serdes` module helps to serialize and deserialize subtypes of Ballerina `anydata`.
```

File: examples/serdes-simple/serdes_simple.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+This sample demonstrates how to serialize and deserialize a user defined record type.<br/><br/>
+For more information on the underlying module,
+see the [Serdes module](https://docs.central.ballerina.io/ballerina/serdes/latest).
```
Comment (praneesha) [inline examples/serdes-simple/serdes_simple.md:6]:
```suggestion
see the [`serdes` module](https://docs.central.ballerina.io/ballerina/serdes/latest).
```

File: examples/serdes-simple/serdes_simple.md (hunk 6 above)
Comment (MohamedSabthar) [reply examples/serdes-simple/serdes_simple.md:3]: Updated

File: examples/serdes-simple/serdes_simple.md (hunk 7 above)
Comment (MohamedSabthar) [reply examples/serdes-simple/serdes_simple.md:6]: Updated

File: examples/serdes-simple/serdes_simple.bal (hunk 5 above)
Comment (MohamedSabthar) [reply examples/serdes-simple/serdes_simple.bal:13]: Updated

File: examples/serdes-simple/serdes_simple.bal (hunk 4 above)
Comment (MohamedSabthar) [reply examples/serdes-simple/serdes_simple.bal:4]: Updated

**Pull request merged**